src/wiki/plugins/images/models.py

```diff
@@ -108,7 +108,12 @@ def on_image_revision_delete(instance, *args, **kwargs):
     if not instance.image:
         return
     # Remove image file
-    path = instance.image.path.split("/")[:-1]
+    try:
+        path = instance.image.path.split("/")[:-1]
+    except NotImplemented:
+            # This backend storage doesn't implement 'path' so there is no path to delete
+        return
```

Comment:
Wait, forgot one thing.. I think you need to move the whole `try ... except` block below the `delete` call. Such that the image is also deleted on S3. I would assume it supports this.